test/jdk/java/lang/annotation/LoaderLeakTest.java

```diff
@@ -72,65 +103,37 @@ static void doTest(boolean readAnn) throws Exception {
     }
 }
 
+@java.lang.annotation.Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
+@interface A {
+    B b();
+}
+
+@interface B { }
+
+@A(b=@B()) class C { }
+
 class SimpleClassLoader extends ClassLoader {
-    private Hashtable classes = new Hashtable();
+    public SimpleClassLoader() { }
 
-    public SimpleClassLoader() {
-    }
-    private byte getClassImplFromDataBase(String className)[] {
+    private byte[] getClassImplFromDataBase(String className) {
         byte result[];
         try {
-            FileInputStream fi = new FileInputStream("classes/"+className+".class");
-            result = new byte[fi.available()];
-            fi.read(result);
-            return result;
+            return Files.readAllBytes(Paths.get(className + ".class"));
         } catch (Exception e) {
-
-            /*
-             * If we caught an exception, either the class wasnt found or it
-             * was unreadable by our process.
-             */
-            return null;
+            throw new Error(e);
```

Comment:
could you please use a descriptive message here (and include `className`  value into it), so it will be easier to analyze test failures?